src/compute-client/src/controller/instance.rs

```diff
@@ -937,7 +937,7 @@ where
                 compute_dependencies.clone(),
                 write_only,
             );
-            updates.push((export_id, replica_write_frontier.clone()));
+            updates.insert(export_id, replica_write_frontier.clone());
```

Comment:
Maybe assert that the id was not un `updates` before.

Reply:
Hm, I think that would be fine actually. If we have duplicate exports IDs (which the definition of `DataflowDescription` precludes, but still) this will just deduplicate them with no harm done. What's not fine is calling `add_collection` for the same ID multiple times, because it changes introspection state, so that should probably panic.

Reply:
Made the `add_collection` methods panic on attempts to add a duplicate ID.